### ISSUE TITLE
When go2rtc self-requests a source capable of backchannel, it breaks backchannel

### ISSUE THREAD

**Issue by felipecrs** [AlexxIT/go2rtc#1133]
It was very tricky to figure out that this was actually caused by go2rtc.

Here is my config:

```yaml
streams:
  vto:
    # H264, PCMA/8000, 2-way audio
    - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
    # copy AAC track from vto_hd
    - rtsp://127.0.0.1:8554/vto_hd?audio=aac
  vto_hd:
    # H264, AAC/16000
    - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=1
    # copy PCMA track from vto
    - rtsp://127.0.0.1:8554/vto?audio=pcma
```

In fact, I have a script to output the stream rather than using it directly because of https://github.com/AlexxIT/go2rtc/issues/49, but I am removing it from this test to simplify things.

Also, my password is leaked but that is ok, I changed already. That's because I could not figure out how to remove it from the info API from go2rtc so it would appear in the video anyway.

Here is the problem:

https://github.com/AlexxIT/go2rtc/assets/29582865/7fb1876a-9e18-4005-a06a-e07f26ad5ba9

Basically, if I open `vto` first, everything works ok. I can 2-way audio with it normally. Notice the sendonly track.

However, if I open `vto_hd` first, then I can no more do 2-way audio with the `vto` stream.

I think go2rtc should attempt to discover again more tracks if 2-way audio is requested. In fact, I don't understand why it only discovers the recvonly tracks when I request `vto_hd`.

**Comment by felipecrs** [comment]
I am working around it at the moment with:

```
streams:
  vto:
    # H264, PCMA/8000, 2-way audio
    - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
    # copy AAC track from vto_hd
    - rtsp://127.0.0.1:8554/vto_hd?audio=aac
  vto_hd:
    # H264, AAC/16000
    - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=1
    # copy PCMA track from vto
    # - rtsp://127.0.0.1:8554/vto?audio=pcma
    - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=0#media=audio
```

**Comment by TokarevSergey** [comment]
> Было очень сложно понять, что на самом деле это вызвано go2rtc.
> 
> Вот моя конфигурация:
> 
> ```yaml
> streams:
>   vto:
>     # H264, PCMA/8000, 2-way audio
>     - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
>     # copy AAC track from vto_hd
>     - rtsp://127.0.0.1:8554/vto_hd?audio=aac
>   vto_hd:
>     # H264, AAC/16000
>     - rtsp://admin:8EiPuNd6pzjR686_@192.168.1.40/cam/realmonitor?channel=1&subtype=1
>     # copy PCMA track from vto
>     - rtsp://127.0.0.1:8554/vto?audio=pcma
> ```
> 
> На самом деле у меня есть скрипт для вывода потока, а не для его прямого использования из-за #49 , но я удаляю его из этого теста, чтобы упростить задачу.
> 
> Кроме того, мой пароль утек, но это нормально, я уже сменил. Это потому, что я не смог придумать, как убрать его из info API от go2rtc, чтобы оно все равно появлялось в видео.
> 
> Вот проблема:
> 
>  chrome_tLNTWxaf4t.mp4 
> В принципе, если сначала открыть `vto`, все работает нормально. Я могу нормально использовать двустороннюю аудиосвязь. Обратите внимание на трек sendonly.
> 
> Однако, если я открою `vto_hd`сначала, я больше не смогу использовать двустороннюю аудиосвязь с `vto`потоком.
> 
> Я думаю, что go2rtc должен попытаться снова обнаружить больше дорожек, если запрошена двусторонняя аудиосвязь. На самом деле, я не понимаю, почему он обнаруживает только треки, когда я запрашиваю `vto_hd`.

Good afternoon.  Tell me, does your sound become quieter and interrupted?

**Comment by felipecrs** [comment]
go2rtc simply doesn't detect the backchannel (sendonly) track, and that's why 2-way audio doesn't work. Whatever other symptom is irrelevant.

**Comment by felipecrs** [comment]
BTW I'm testing this more and even with the workaround, sometimes go2rtc simply "forgets" about the `sendonly` track. I still can't figure out why. But if I restart go2rtc, then it catches the `sendonly` track again.

**Comment by felipecrs** [comment]
Hm... I actually understand now. When someone calls the doorbell, like when someone presses it, the doorbell closes the RTSP stream momentarily. And go2rtc tries to get the connection again but this time it gets without `sendonly` track. Only if I ask go2rtc to get the connection after the call is finished it will be able to get `sendonly` again.

**Comment by TokarevSergey** [comment]
Yes, yes, I’ve already tried to write this several times.

**Comment by felipecrs** [comment]
Another option is maybe https://github.com/AlexxIT/go2rtc/issues/52, maybe it doesn't have the same limitation.

**Comment by AlexxIT** [comment]
RTSP backchannel has two fallbacks in case of any problems. You can increase log level for rtsp module to trace.

Check this:
https://github.com/AlexxIT/go2rtc/blob/686fb374e960646a7c52a976289023534989ee9d/pkg/rtsp/client.go#L241-L249

And this:
https://github.com/AlexxIT/go2rtc/blob/686fb374e960646a7c52a976289023534989ee9d/internal/rtsp/rtsp.go#L126-L140

**Comment by TokarevSergey** [comment]
How will he fix it? So that everything works.

**Comment by AlexxIT** [comment]
1. You take the problem device and throw it in the trash.
2. Buy a new device that doesn't have these firmware issues.

Or you just accept your device's firmware problems.

RTSP backchannel is a new technology for camera developers. My new Dahua camera has no such problems.

**Comment by felipecrs** [comment]
> RTSP backchannel has two fallbacks in case of any problems. You can increase log level for rtsp module to trace.

Thank you, I will collect this information to continue the investigation.

In fact I am causing some of this confusion because I was mentioning several issues after working around the first.

I will focus on the first problem I mentioned, which is the issue description. This one does not appear to be caused by the device itself.

Therefore, @AlexxIT, I think it may be possible to replicate the issue using your current Dahua camera. But let me do things on my side first.

Also, @TokarevSergey, AFAIK Hikvision works fine using ISAPI. There was even a fix to ensure connections are closed recently in go2rtc. Regardless, let's try to keep this topic straight to a single issue. If you have something not working as expected with Hikvision, I suggest you open a different issue.

**Comment by TokarevSergey** [comment]
> > Обратный канал RTSP имеет два резерва на случай каких-либо проблем. Вы можете повысить уровень журнала для отслеживания модуля rtsp.
> 
> Спасибо, я соберу эту информацию, чтобы продолжить расследование.
> 
> На самом деле я вношу некоторую путаницу, потому что упомянул несколько проблем после работы над первой.
> 
> Я сосредоточусь на первой проблеме, которую я упомянул, а именно на ее описании. Похоже, что это не вызвано самим устройством.
> 
> Поэтому,@AlexxIT, Я думаю, что можно воспроизвести проблему, используя вашу текущую камеру Dahua. Но позвольте мне сначала сделать что-то со своей стороны.
> 
> Также,@TokarevSergey, AFAIK Hikvision отлично работает с использованием ISAPI. Недавно было даже исправлено, гарантирующее закрытие соединений в go2rtc. В любом случае, давайте попробуем свести эту тему к одному вопросу. Если у вас что-то не работает с Hikvision, я предлагаю вам открыть другую проблему.

Thanks, I've already opened a bunch of topics. Perhaps I lack the experience to explain all this correctly.

**Comment by AlexxIT** [comment]
trace. Not debug

**Comment by felipecrs** [comment]
Ops. Here is the trace:

[logs.txt](https://github.com/AlexxIT/go2rtc/files/15420211/logs.txt)

**Comment by felipecrs** [comment]
And here is the trace if I just open `vto` without opening `vto_hd` first:


[logs.txt](https://github.com/AlexxIT/go2rtc/files/15420220/logs.txt)

For comparison.

**Comment by AlexxIT** [comment]
Well. `vto_hd` should have `#backchannel=0`. Because this request is made first and it takes over the microphone.

1
```
5/23/2024, 2:05:24 PM	trace	[rtsp] client request:
DESCRIBE rtsp://192.168.1.40/cam/realmonitor?channel=1&subtype=1 RTSP/1.0
Require: www.onvif.org/ver20/backchannel

5/23/2024, 2:05:24 PM	trace	[rtsp] client response:
RTSP/1.0 200 OK
```

2
```
5/23/2024, 2:05:24 PM	trace	[rtsp] client request:
DESCRIBE rtsp://192.168.1.40/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif RTSP/1.0
Require: www.onvif.org/ver20/backchannel

5/23/2024, 2:05:25 PM	trace	[rtsp] client response:
RTSP/1.0 404 Not Found

5/23/2024, 2:05:25 PM	trace	[rtsp] describe (backchannel=true) err: wrong response on DESCRIBE
```

**Comment by felipecrs** [comment]
It worked! And it makes total sense. So here's the working config:

```yaml
api:
  origin: '*'
log:
  rtsp: debug
streams:
  vto:
    # H264, PCMA/8000, 2-way audio
    - rtsp://admin:pass@192.168.1.40/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
    # copy AAC track from vto_hd
    - rtsp://127.0.0.1:8554/vto_hd?audio=aac
  vto_hd:
    # H264, AAC/16000
    - rtsp://admin:pass@192.168.1.40/cam/realmonitor?channel=1&subtype=1#backchannel=0
    # copy PCMA track from vto
    - rtsp://127.0.0.1:8554/vto?audio=pcma
webrtc:
  candidates:
  - 192.168.1.10:8555
  - stun:8555
```

Thanks a lot for the patience.
